aten/src/ATen/native/cuda/ROCmWorkaround.cuh

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+namespace rocm { namespace workaround {
+
+template<typename T>
+struct enable_default_constructor {
+  struct { char bytes[sizeof(T)]; } value;
```

Comment:
This should have `static_assert(std::is_trivially_copyable<T>{} && std::is_trivially_destructible<T>{}, "Not trivial type")` to make sure this isn't used with types that are not trivially copyable.

Reply:
fixed

Reply:
This doesn't work either. I think the problem is `thrust::complex` should be trivially constructible/copyable/destructible but it is not declared so in ROCm, therefore we need this workaround to bypass this compiler check.